CARET_trace/src/ros_trace_points.cpp

```diff
@@ -51,6 +52,75 @@
 std::unique_ptr<std::thread> trace_node_thread;
 thread_local bool trace_filter_is_rcl_publish_recorded;
 
+using std::string;
+using std::vector;
+static bool ignore_rcl_timer_init = false;
+
+static vector<string> string_split(string & str, char delim)
+{
+  using std::stringstream;
+  vector<string> elems;
+  stringstream ss(str);
+  string item;
+  while (getline(ss, item, delim)) {
+    if (!item.empty()) {
+      elems.push_back(item);
+    }
+  }
+  return elems;
+}
+
+static bool is_python3_command()
+{
+  using std::ifstream;
+  using std::ios;
+  using std::istreambuf_iterator;
+  using std::to_string;
+  pid_t pid = getpid();
+  string fileName = "/proc/" + to_string(pid) + "/cmdline";
+
+  ifstream ifs(fileName, ios::in | ios::binary);
+  if (!ifs) {
+    return false;
+  }
+  istreambuf_iterator<char> it_ifs_begin(ifs);
+  istreambuf_iterator<char> it_ifs_end{};
+  vector<char> input_data(it_ifs_begin, it_ifs_end);
+  if (!ifs) {
+    ifs.close();
+    return false;
+  }
+  ifs.close();
+
+  vector<string> cmd_line;
+  auto itr_begin = input_data.begin();
+  auto itr_find = input_data.begin();
+  for (;;) {
+    itr_find = find(itr_begin, input_data.end(), 0x00);
+    if (itr_find == input_data.end()) {
+      break;
+    }
+    string output_string(itr_begin, itr_find);
+    cmd_line.push_back(output_string);
+    itr_begin = itr_find + 1;
+  }
+
+  // Check if the right side of the last '/' in cmdline[0] matches 'python3'.
+  if (cmd_line.size() < 1) {
+    return false;
+  }
+  // case 1 : /usr/bin/python3 /opt/ros/humble/bin/ros2 launch ...
+  // case 2 :  python3 /opt/ros/humble/lib/rosbridge_server/rosbridge_websocket ...
+  if (cmd_line[0].find("python3") == string::npos) {
```

Comment:
I just would like to confirm it, but do we need this if block? L117-L120 seems to have the same logic.
Or, do you want to avoid calling string_split (heavy logic?) to shorten processing time?

Reply:
I don't think it's a big difference, but in the case of autoware, it will be called a lot of times, so find will finish in a shorter time, so I tried not to call string_split as much as possible.